server/src/main/java/io/crate/planner/node/ddl/AlterTableAddColumnPlan.java

```diff
@@ -78,25 +90,60 @@ public void executeOrFail(DependencyCarrier dependencies,
                               RowConsumer consumer,
                               Row params,
                               SubQueryResults subQueryResults) throws Exception {
-        BoundAddColumn stmt = bind(
-            alterTable,
-            plannerContext.transactionContext(),
-            dependencies.nodeContext(),
-            params,
-            subQueryResults,
-            dependencies.fulltextAnalyzerResolver());
+        if (plannerContext.clusterState().getNodes().getMinNodeVersion().onOrAfter(Version.V_5_1_0)) {
```

Comment:
as discussed, we are going to forbid DDL on mixed clusters. But ADD COLUMN would be an exception as we allow DML on mixed clusters ==> dynamic mapping updates implicitly causes DDL (col additions) so I kept version check here